File: data/data.code.lua53.server/ReqTestJs.lua
ReqTestJs = {}
setmetatable(ReqTestJs, {__index = _G})
local _ENV = ReqTestJs


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqTestJs})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.u64 = 0
	self.i64 = 0
end


function decode(self, pack)
	self.u64 = pack:readu64()
	self.i64 = pack:readi64()
	return self
end


function getU64(self)
	return self.u64
end

function getI64(self)
	return self.i64
end

File: data/data.code.cpp.server/ReqTestJs.cpp
#ifndef _REQ_TEST_JS_
#define _REQ_TEST_JS_

#include <list>

#include <pb_type.hpp>
#include <Packet.hpp>

#include <Msg.hpp>



class ReqTestJs
{
private:
	U64 m_u64;
	I64 m_i64;


public:
	ReqTestJs(Packet* packet)
	{
		m_u64 = packet->ReadU64();
		m_i64 = packet->ReadI64();
	}


	U64 GetU64()
	{
		return m_u64;
	}

	I64 GetI64()
	{
		return m_i64;
	}

};

#endif
